File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/AgentFlowSession.java
package com.oracle.aidataplatform.dp.model;



/**
 * Agent flow session.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=AgentFlowSession.Builder.class)

public final class AgentFlowSession  {
    @Deprecated
    @java.beans.ConstructorProperties({"displayName", "key", "agentFlowKey", "endpointUrl", "lifecycleState", "agentFlowComputeKey", "timeCreated", "updatedBy", "createdBy", "timeUpdated", "timeStarted", "timeEnded", "context"})
    public AgentFlowSession(String displayName, String key, String agentFlowKey, String endpointUrl, DeploymentLifecycleState lifecycleState, String agentFlowComputeKey, java.util.Date timeCreated, String updatedBy, String createdBy, java.util.Date timeUpdated, java.util.Date timeStarted, java.util.Date timeEnded, AgentFlowSessionContext context) {
        super();
        this.displayName = displayName;
        this.key = key;
        this.agentFlowKey = agentFlowKey;
        this.endpointUrl = endpointUrl;
        this.lifecycleState = lifecycleState;
        this.agentFlowComputeKey = agentFlowComputeKey;
        this.timeCreated = timeCreated;
        this.updatedBy = updatedBy;
        this.createdBy = createdBy;
        this.timeUpdated = timeUpdated;
        this.timeStarted = timeStarted;
        this.timeEnded = timeEnded;
        this.context = context;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Display name.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

        /**
         * Display name.
         * @param displayName the value to set
         * @return this builder
         **/
        

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
            /**
     * Agent Flow Session identifier.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * Agent Flow Session identifier.
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * The Agent Flow key for which the session is started.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
private String agentFlowKey;

        /**
         * The Agent Flow key for which the session is started.
         * @param agentFlowKey the value to set
         * @return this builder
         **/
        

public Builder agentFlowKey(String agentFlowKey) {
    this.agentFlowKey = agentFlowKey;
    return this;
}
            /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("endpointUrl")
private String endpointUrl;

        /**
         * The endpointUrl where the client should connect to communicate with the Agent.
         * @param endpointUrl the value to set
         * @return this builder
         **/
        

public Builder endpointUrl(String endpointUrl) {
    this.endpointUrl = endpointUrl;
    return this;
}
            /**
     * LifecycleState of an Agent Flow Session or Deployment.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private DeploymentLifecycleState lifecycleState;

        /**
         * LifecycleState of an Agent Flow Session or Deployment.
         * @param lifecycleState the value to set
         * @return this builder
         **/
        

public Builder lifecycleState(DeploymentLifecycleState lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}
            /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("agentFlowComputeKey")
private String agentFlowComputeKey;

        /**
         * The Agent Flow Compute Key where client can run or test the Agent Flow.
         * @param agentFlowComputeKey the value to set
         * @return this builder
         **/
        

public Builder agentFlowComputeKey(String agentFlowComputeKey) {
    this.agentFlowComputeKey = agentFlowComputeKey;
    return this;
}
            /**
     * The date and time the Agent flow session was created.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * The date and time the Agent flow session was created.
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * OCID of the user who updated this record
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
private String updatedBy;

        /**
         * OCID of the user who updated this record
         * @param updatedBy the value to set
         * @return this builder
         **/
        

public Builder updatedBy(String updatedBy) {
    this.updatedBy = updatedBy;
    return this;
}
            /**
     * The OCID of the user/principal who created the Agent flow session.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

        /**
         * The OCID of the user/principal who created the Agent flow session.
         * @param createdBy the value to set
         * @return this builder
         **/
        

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
            /**
     * The date and time the Agent flow session was updated.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

        /**
         * The date and time the Agent flow session was updated.
         * @param timeUpdated the value to set
         * @return this builder
         **/
        

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
            /**
     * The date and time the session was started
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeStarted")
private java.util.Date timeStarted;

        /**
         * The date and time the session was started
         * @param timeStarted the value to set
         * @return this builder
         **/
        

public Builder timeStarted(java.util.Date timeStarted) {
    this.timeStarted = timeStarted;
    return this;
}
            /**
     * The date and time the session was ended
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeEnded")
private java.util.Date timeEnded;

        /**
         * The date and time the session was ended
         * @param timeEnded the value to set
         * @return this builder
         **/
        

public Builder timeEnded(java.util.Date timeEnded) {
    this.timeEnded = timeEnded;
    return this;
}
        
@com.fasterxml.jackson.annotation.JsonProperty("context")
private AgentFlowSessionContext context;



public Builder context(AgentFlowSessionContext context) {
    this.context = context;
    return this;
}


        public AgentFlowSession build() {
            AgentFlowSession model = new AgentFlowSession(this.displayName
                , this.key
                , this.agentFlowKey
                , this.endpointUrl
                , this.lifecycleState
                , this.agentFlowComputeKey
                , this.timeCreated
                , this.updatedBy
                , this.createdBy
                , this.timeUpdated
                , this.timeStarted
                , this.timeEnded
                , this.context);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(AgentFlowSession model) {
                this.displayName(model.getDisplayName());
    this.key(model.getKey());
    this.agentFlowKey(model.getAgentFlowKey());
    this.endpointUrl(model.getEndpointUrl());
    this.lifecycleState(model.getLifecycleState());
    this.agentFlowComputeKey(model.getAgentFlowComputeKey());
    this.timeCreated(model.getTimeCreated());
    this.updatedBy(model.getUpdatedBy());
    this.createdBy(model.getCreatedBy());
    this.timeUpdated(model.getTimeUpdated());
    this.timeStarted(model.getTimeStarted());
    this.timeEnded(model.getTimeEnded());
    this.context(model.getContext());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Display name.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * Display name.
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * Agent Flow Session identifier.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * Agent Flow Session identifier.
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * The Agent Flow key for which the session is started.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowKey")
    private final String agentFlowKey;

        /**
     * The Agent Flow key for which the session is started.
     * @return the value
     **/
    
    public String getAgentFlowKey() {
        return agentFlowKey;
    }


        /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("endpointUrl")
    private final String endpointUrl;

        /**
     * The endpointUrl where the client should connect to communicate with the Agent.
     * @return the value
     **/
    
    public String getEndpointUrl() {
        return endpointUrl;
    }

    
        /**
     * LifecycleState of an Agent Flow Session or Deployment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
    private final DeploymentLifecycleState lifecycleState;

        /**
     * LifecycleState of an Agent Flow Session or Deployment.
     * @return the value
     **/
    
    public DeploymentLifecycleState getLifecycleState() {
        return lifecycleState;
    }


        /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("agentFlowComputeKey")
    private final String agentFlowComputeKey;

        /**
     * The Agent Flow Compute Key where client can run or test the Agent Flow.
     * @return the value
     **/
    
    public String getAgentFlowComputeKey() {
        return agentFlowComputeKey;
    }


        /**
     * The date and time the Agent flow session was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * The date and time the Agent flow session was created.
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * OCID of the user who updated this record
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
    private final String updatedBy;

        /**
     * OCID of the user who updated this record
     * @return the value
     **/
    
    public String getUpdatedBy() {
        return updatedBy;
    }


        /**
     * The OCID of the user/principal who created the Agent flow session.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
    private final String createdBy;

        /**
     * The OCID of the user/principal who created the Agent flow session.
     * @return the value
     **/
    
    public String getCreatedBy() {
        return createdBy;
    }


        /**
     * The date and time the Agent flow session was updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

        /**
     * The date and time the Agent flow session was updated.
     * @return the value
     **/
    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }


        /**
     * The date and time the session was started
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeStarted")
    private final java.util.Date timeStarted;

        /**
     * The date and time the session was started
     * @return the value
     **/
    
    public java.util.Date getTimeStarted() {
        return timeStarted;
    }


        /**
     * The date and time the session was ended
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeEnded")
    private final java.util.Date timeEnded;

        /**
     * The date and time the session was ended
     * @return the value
     **/
    
    public java.util.Date getTimeEnded() {
        return timeEnded;
    }


    
    @com.fasterxml.jackson.annotation.JsonProperty("context")
    private final AgentFlowSessionContext context;

    
    public AgentFlowSessionContext getContext() {
        return context;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("AgentFlowSession(");
        sb.append("displayName=").append(String.valueOf(this.displayName));
        sb.append(", key=").append(String.valueOf(this.key));
        sb.append(", agentFlowKey=").append(String.valueOf(this.agentFlowKey));
        sb.append(", endpointUrl=").append(String.valueOf(this.endpointUrl));
        sb.append(", lifecycleState=").append(String.valueOf(this.lifecycleState));
        sb.append(", agentFlowComputeKey=").append(String.valueOf(this.agentFlowComputeKey));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", updatedBy=").append(String.valueOf(this.updatedBy));
        sb.append(", createdBy=").append(String.valueOf(this.createdBy));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(", timeStarted=").append(String.valueOf(this.timeStarted));
        sb.append(", timeEnded=").append(String.valueOf(this.timeEnded));
        sb.append(", context=").append(String.valueOf(this.context));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof AgentFlowSession)) {
            return false;
        }

        AgentFlowSession other = (AgentFlowSession) o;
        return java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.agentFlowKey, other.agentFlowKey) &&
            java.util.Objects.equals(this.endpointUrl, other.endpointUrl) &&
            java.util.Objects.equals(this.lifecycleState, other.lifecycleState) &&
            java.util.Objects.equals(this.agentFlowComputeKey, other.agentFlowComputeKey) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.updatedBy, other.updatedBy) &&
            java.util.Objects.equals(this.createdBy, other.createdBy) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated) &&
            java.util.Objects.equals(this.timeStarted, other.timeStarted) &&
            java.util.Objects.equals(this.timeEnded, other.timeEnded) &&
            java.util.Objects.equals(this.context, other.context);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.agentFlowKey == null ? 43 : this.agentFlowKey.hashCode());
        result = (result * PRIME) + (this.endpointUrl == null ? 43 : this.endpointUrl.hashCode());
        result = (result * PRIME) + (this.lifecycleState == null ? 43 : this.lifecycleState.hashCode());
        result = (result * PRIME) + (this.agentFlowComputeKey == null ? 43 : this.agentFlowComputeKey.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.updatedBy == null ? 43 : this.updatedBy.hashCode());
        result = (result * PRIME) + (this.createdBy == null ? 43 : this.createdBy.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        result = (result * PRIME) + (this.timeStarted == null ? 43 : this.timeStarted.hashCode());
        result = (result * PRIME) + (this.timeEnded == null ? 43 : this.timeEnded.hashCode());
        result = (result * PRIME) + (this.context == null ? 43 : this.context.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/agent_flow_session.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class AgentFlowSession(object):
    """
    Agent flow session.
    """

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSession.
    #: This constant has a value of "CREATING"
    LIFECYCLE_STATE_CREATING = "CREATING"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSession.
    #: This constant has a value of "ACTIVE"
    LIFECYCLE_STATE_ACTIVE = "ACTIVE"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSession.
    #: This constant has a value of "INACTIVE"
    LIFECYCLE_STATE_INACTIVE = "INACTIVE"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSession.
    #: This constant has a value of "FAILED"
    LIFECYCLE_STATE_FAILED = "FAILED"

    #: A constant which can be used with the lifecycle_state property of a AgentFlowSession.
    #: This constant has a value of "DELETED"
    LIFECYCLE_STATE_DELETED = "DELETED"

    def __init__(self, **kwargs):
        """
        Initializes a new AgentFlowSession object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param display_name:
            The value to assign to the display_name property of this AgentFlowSession.
        :type display_name: str

        :param key:
            The value to assign to the key property of this AgentFlowSession.
        :type key: str

        :param agent_flow_key:
            The value to assign to the agent_flow_key property of this AgentFlowSession.
        :type agent_flow_key: str

        :param endpoint_url:
            The value to assign to the endpoint_url property of this AgentFlowSession.
        :type endpoint_url: str

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this AgentFlowSession.
            Allowed values for this property are: "CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"
        :type lifecycle_state: str

        :param agent_flow_compute_key:
            The value to assign to the agent_flow_compute_key property of this AgentFlowSession.
        :type agent_flow_compute_key: str

        :param time_created:
            The value to assign to the time_created property of this AgentFlowSession.
        :type time_created: datetime

        :param updated_by:
            The value to assign to the updated_by property of this AgentFlowSession.
        :type updated_by: str

        :param created_by:
            The value to assign to the created_by property of this AgentFlowSession.
        :type created_by: str

        :param time_updated:
            The value to assign to the time_updated property of this AgentFlowSession.
        :type time_updated: datetime

        :param time_started:
            The value to assign to the time_started property of this AgentFlowSession.
        :type time_started: datetime

        :param time_ended:
            The value to assign to the time_ended property of this AgentFlowSession.
        :type time_ended: datetime

        :param context:
            The value to assign to the context property of this AgentFlowSession.
        :type context: oci.aidataplatform_dp.models.AgentFlowSessionContext

        """
        self.swagger_types = {
            'display_name': 'str',
            'key': 'str',
            'agent_flow_key': 'str',
            'endpoint_url': 'str',
            'lifecycle_state': 'str',
            'agent_flow_compute_key': 'str',
            'time_created': 'datetime',
            'updated_by': 'str',
            'created_by': 'str',
            'time_updated': 'datetime',
            'time_started': 'datetime',
            'time_ended': 'datetime',
            'context': 'AgentFlowSessionContext'
        }

        self.attribute_map = {
            'display_name': 'displayName',
            'key': 'key',
            'agent_flow_key': 'agentFlowKey',
            'endpoint_url': 'endpointUrl',
            'lifecycle_state': 'lifecycleState',
            'agent_flow_compute_key': 'agentFlowComputeKey',
            'time_created': 'timeCreated',
            'updated_by': 'updatedBy',
            'created_by': 'createdBy',
            'time_updated': 'timeUpdated',
            'time_started': 'timeStarted',
            'time_ended': 'timeEnded',
            'context': 'context'
        }

        self._display_name = None
        self._key = None
        self._agent_flow_key = None
        self._endpoint_url = None
        self._lifecycle_state = None
        self._agent_flow_compute_key = None
        self._time_created = None
        self._updated_by = None
        self._created_by = None
        self._time_updated = None
        self._time_started = None
        self._time_ended = None
        self._context = None

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this AgentFlowSession.
        Display name.


        :return: The display_name of this AgentFlowSession.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this AgentFlowSession.
        Display name.


        :param display_name: The display_name of this AgentFlowSession.
        :type: str
        """
        self._display_name = display_name

    @property
    def key(self):
        """
        **[Required]** Gets the key of this AgentFlowSession.
        Agent Flow Session identifier.


        :return: The key of this AgentFlowSession.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this AgentFlowSession.
        Agent Flow Session identifier.


        :param key: The key of this AgentFlowSession.
        :type: str
        """
        self._key = key

    @property
    def agent_flow_key(self):
        """
        **[Required]** Gets the agent_flow_key of this AgentFlowSession.
        The Agent Flow key for which the session is started.


        :return: The agent_flow_key of this AgentFlowSession.
        :rtype: str
        """
        return self._agent_flow_key

    @agent_flow_key.setter
    def agent_flow_key(self, agent_flow_key):
        """
        Sets the agent_flow_key of this AgentFlowSession.
        The Agent Flow key for which the session is started.


        :param agent_flow_key: The agent_flow_key of this AgentFlowSession.
        :type: str
        """
        self._agent_flow_key = agent_flow_key

    @property
    def endpoint_url(self):
        """
        **[Required]** Gets the endpoint_url of this AgentFlowSession.
        The endpointUrl where the client should connect to communicate with the Agent.


        :return: The endpoint_url of this AgentFlowSession.
        :rtype: str
        """
        return self._endpoint_url

    @endpoint_url.setter
    def endpoint_url(self, endpoint_url):
        """
        Sets the endpoint_url of this AgentFlowSession.
        The endpointUrl where the client should connect to communicate with the Agent.


        :param endpoint_url: The endpoint_url of this AgentFlowSession.
        :type: str
        """
        self._endpoint_url = endpoint_url

    @property
    def lifecycle_state(self):
        """
        **[Required]** Gets the lifecycle_state of this AgentFlowSession.
        LifecycleState of an Agent Flow Session or Deployment.

        Allowed values for this property are: "CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"


        :return: The lifecycle_state of this AgentFlowSession.
        :rtype: str
        """
        return self._lifecycle_state

    @lifecycle_state.setter
    def lifecycle_state(self, lifecycle_state):
        """
        Sets the lifecycle_state of this AgentFlowSession.
        LifecycleState of an Agent Flow Session or Deployment.


        :param lifecycle_state: The lifecycle_state of this AgentFlowSession.
        :type: str
        """
        allowed_values = ["CREATING", "ACTIVE", "INACTIVE", "FAILED", "DELETED"]
        if not value_allowed_none_or_none_sentinel(lifecycle_state, allowed_values):
            raise ValueError(
                "Invalid value for `lifecycle_state`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._lifecycle_state = lifecycle_state

    @property
    def agent_flow_compute_key(self):
        """
        **[Required]** Gets the agent_flow_compute_key of this AgentFlowSession.
        The Agent Flow Compute Key where client can run or test the Agent Flow.


        :return: The agent_flow_compute_key of this AgentFlowSession.
        :rtype: str
        """
        return self._agent_flow_compute_key

    @agent_flow_compute_key.setter
    def agent_flow_compute_key(self, agent_flow_compute_key):
        """
        Sets the agent_flow_compute_key of this AgentFlowSession.
        The Agent Flow Compute Key where client can run or test the Agent Flow.


        :param agent_flow_compute_key: The agent_flow_compute_key of this AgentFlowSession.
        :type: str
        """
        self._agent_flow_compute_key = agent_flow_compute_key

    @property
    def time_created(self):
        """
        Gets the time_created of this AgentFlowSession.
        The date and time the Agent flow session was created.


        :return: The time_created of this AgentFlowSession.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this AgentFlowSession.
        The date and time the Agent flow session was created.


        :param time_created: The time_created of this AgentFlowSession.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def updated_by(self):
        """
        Gets the updated_by of this AgentFlowSession.
        OCID of the user who updated this record


        :return: The updated_by of this AgentFlowSession.
        :rtype: str
        """
        return self._updated_by

    @updated_by.setter
    def updated_by(self, updated_by):
        """
        Sets the updated_by of this AgentFlowSession.
        OCID of the user who updated this record


        :param updated_by: The updated_by of this AgentFlowSession.
        :type: str
        """
        self._updated_by = updated_by

    @property
    def created_by(self):
        """
        Gets the created_by of this AgentFlowSession.
        The OCID of the user/principal who created the Agent flow session.


        :return: The created_by of this AgentFlowSession.
        :rtype: str
        """
        return self._created_by

    @created_by.setter
    def created_by(self, created_by):
        """
        Sets the created_by of this AgentFlowSession.
        The OCID of the user/principal who created the Agent flow session.


        :param created_by: The created_by of this AgentFlowSession.
        :type: str
        """
        self._created_by = created_by

    @property
    def time_updated(self):
        """
        Gets the time_updated of this AgentFlowSession.
        The date and time the Agent flow session was updated.


        :return: The time_updated of this AgentFlowSession.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this AgentFlowSession.
        The date and time the Agent flow session was updated.


        :param time_updated: The time_updated of this AgentFlowSession.
        :type: datetime
        """
        self._time_updated = time_updated

    @property
    def time_started(self):
        """
        Gets the time_started of this AgentFlowSession.
        The date and time the session was started


        :return: The time_started of this AgentFlowSession.
        :rtype: datetime
        """
        return self._time_started

    @time_started.setter
    def time_started(self, time_started):
        """
        Sets the time_started of this AgentFlowSession.
        The date and time the session was started


        :param time_started: The time_started of this AgentFlowSession.
        :type: datetime
        """
        self._time_started = time_started

    @property
    def time_ended(self):
        """
        Gets the time_ended of this AgentFlowSession.
        The date and time the session was ended


        :return: The time_ended of this AgentFlowSession.
        :rtype: datetime
        """
        return self._time_ended

    @time_ended.setter
    def time_ended(self, time_ended):
        """
        Sets the time_ended of this AgentFlowSession.
        The date and time the session was ended


        :param time_ended: The time_ended of this AgentFlowSession.
        :type: datetime
        """
        self._time_ended = time_ended

    @property
    def context(self):
        """
        Gets the context of this AgentFlowSession.

        :return: The context of this AgentFlowSession.
        :rtype: oci.aidataplatform_dp.models.AgentFlowSessionContext
        """
        return self._context

    @context.setter
    def context(self, context):
        """
        Sets the context of this AgentFlowSession.

        :param context: The context of this AgentFlowSession.
        :type: oci.aidataplatform_dp.models.AgentFlowSessionContext
        """
        self._context = context

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
